### PR TITLE
Updates for source-code FMUs

### PR DESCRIPTION
**Pull request by sjoelund**
Make it possible to compile source-code FMUs without setting include-
directives to the preprocessor. This required changes to most files
since they assume preprocessor flags were set.

The generated FMU now contains a list of the files to compile (in the
XML structure), in case the user does not have access to a make system.

Still missing is a config-file containing number of linear systems, etc.
This would make it possible to compile without configure-script at all.
The configure script should also look for the standard FMI includes
which should not need to be present in a source-code FMU.

### CODE REVIEW

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2019-03-21_14-40-45](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6528/console).

File: SimulationRuntime/c/RuntimeSources.mo.tpl [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  constant list<String> fmi2Files={"fmi2/fmu2_model_interface.c.inc"};
+  constant list<String> fmi2Headers={"fmi2/fmi2Functions.h","fmi2/fmi2TypesPlatform.h","fmi2/fmu2_model_interface.h","fmi2/fmi2FunctionTypes.h"};
+  constant list<String> fmi2AllFiles=listAppend(fmi2Files, fmi2Headers);
+  constant list<String> defaultFileSuffixes={".c", "_functions.c", "_records.c", "_01exo.c", "_02nls.c", "_03lsy.c", "_04set.c", "_05evt.c", "_06inz.c", "_07dly.c", "_08bnd.c", "_09alg.c", "_10asr.c", "_11mix.c", "_12jac.c", "_13opt.c", "_14lnz.c", "_15syn.c", "_16dae.c", "_17inl.c", "_init_fmu.c", "_FMU.c"};
+  constant list<String> dgesvFiles={DGESV_FILES, "./external_solvers/blaswrap.h", "./external_solvers/clapack.h", "./external_solvers/f2c.h"};
+  constant list<String> lsFiles={LS_FILES};
+  constant list<String> nlsFiles={NLS_FILESCMINPACK_FILES, "./external_solvers/cminpack.h", "./external_solvers/minpack.h"};
+  constant list<String> mixedFiles={MIXED_FILES};
+annotation(__OpenModelica_Interface="backend");
+end RuntimeSources;
```
Comment (adeas31) [inline SimulationRuntime/c/RuntimeSources.mo.tpl]: What about the header files like model_literals.h, model_includes.h etc.? I don't see them being copied from somewhere.

Comment (sjoelund) [reply SimulationRuntime/c/RuntimeSources.mo.tpl]: They are generated directly in the correct location. Then they are put in the XML using defaultFileSuffixes

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2019-03-25_16-38-59](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6534/console).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2019-03-26_13-13-45](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6543/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2019-03-26_14-45-10](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6546/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2019-03-27_08-07-16](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6548/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2019-03-27_08-26-30](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6549/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2019-03-27_14-51-32](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6550/).

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2019-03-27_15-42-04](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/6552/console).

**Pull request closed without merging**